### ISSUE TITLE
Reported Syntax error when using conditional facts["${facts['factname']}"] 

### ISSUE THREAD

**Issue by Cinderhaze** [puppetlabs/puppet-lint#154]
## Describe the Bug
We are trying to use the value of a fact as a key for looking up another fact in a conditional.  Inside the body of the conditional, we are trying to use the fact, and it is reporting as a syntax error, but it passes `puppet parser validate`

Initial example that caused it to fail
```
if $facts["file_${facts['some_custom_fact']}_yaml] {
  service: "${facts['some_custom_fact']}":
    ensure => running,
    enable => true
}
```
Error was on the line inside of the conditional where we were setting the service name.

I noticed that if I commented out the `service` line and made changes below it, I still got syntax errors on the commented line.
```
if $facts["file_${facts['some_custom_fact']}_yaml] {
  service: "$some_custom_fact}":
#  service: "${facts['some_custom_fact']}":
    ensure => running,
    enable => true
}
```

## Expected Behavior
Given usage of using a facts[] lookup as part of a facts lookup is valid consistent with `puppet parser validate`

## Steps to Reproduce
```
cat << 'EOF' > error.pp
if $facts["[]"] {
  #"${bar}"
}
EOF
puppet-lint error.pp

## Environment
 - Version [2.5.2, 4.0.0(pdk)]

## Additional Context
Minimal example that still signals syntax error seems to occur if there are square brackets in the conditional string.  Note that the line that it fails on is the comment inside the 
```
if $facts["[]"] {
  #"${bar}"
}
```

This leads me to believe that there is some error in parsing the 'key' in the facts map, that causes some kind of downstream problems.

We can workaround this by storing the fact in a local variable, and using the local variable for any references within our class.
